storagemarket/impl/providerstates/provider_states.go

```diff
@@ -341,17 +345,26 @@ func HandoffDeal(ctx fsm.Context, environment ProviderDealEnvironment, deal stor
 		_ = ctx.Trigger(storagemarket.ProviderEventPieceStoreErrored, err)
 	}
 
-	if err := environment.ActivateShard(deal.Proposal.PieceCID); err != nil {
+	if err := environment.RegisterShard(ctx.Context(), deal.Proposal.PieceCID, carFilePath); err != nil {
 		// TODO What's the right thing to do here ? I think the retrieval market
```

Comment:
We can remove this TODO.  We will now do[ Shard recovery ](https://github.com/filecoin-project/dagstore/issues/33) inside the DAG Store Wrapper's `RegisterShard `.